src/js/events/MetronomeActions.js

```diff
@@ -22,201 +20,134 @@ import {
 } from "../constants/ActionTypes";
 import AppDispatcher from "./AppDispatcher";
 import Config from "../config/Config";
+import * as MetronomeClient from "./MetronomeClient";
 
 const MetronomeActions = {
   createJob(data) {
-    RequestUtil.json({
-      url: `${Config.metronomeAPI}/v0/scheduled-jobs`,
-      method: "POST",
-      data,
-      success() {
+    MetronomeClient.createJob(data).subscribe({
+      next: () =>
         AppDispatcher.handleServerAction({
           type: REQUEST_METRONOME_JOB_CREATE_SUCCESS
-        });
-      },
-      error(xhr) {
+        }),
+      error: xhr =>
         AppDispatcher.handleServerAction({
           type: REQUEST_METRONOME_JOB_CREATE_ERROR,
           data: RequestUtil.parseResponseBody(xhr),
           xhr
-        });
-      }
+        })
     });
   },
 
-  fetchJobs: RequestUtil.debounceOnError(
-    Config.getRefreshRate(),
-    function(resolve, reject) {
-      return function() {
-        RequestUtil.json({
-          url: `${Config.metronomeAPI}/v1/jobs`,
-          data: [
-            { name: "embed", value: "activeRuns" },
-            { name: "embed", value: "schedules" },
-            { name: "embed", value: "historySummary" }
-          ],
-          success(response) {
-            AppDispatcher.handleServerAction({
-              type: REQUEST_METRONOME_JOBS_SUCCESS,
-              data: response
-            });
-          },
-          error(xhr) {
-            AppDispatcher.handleServerAction({
-              type: REQUEST_METRONOME_JOBS_ERROR,
-              data: xhr.message,
-              xhr
-            });
-            reject();
-          },
-          hangingRequestCallback() {
-            AppDispatcher.handleServerAction({
-              type: REQUEST_METRONOME_JOBS_ONGOING
-            });
-          }
-        });
-      };
-    },
-    { delayAfterCount: Config.delayAfterErrorCount }
-  ),
+  fetchJobs() {
+    MetronomeClient.fetchJobs().subscribe({
+      next: data =>
+        AppDispatcher.handleServerAction({
+          type: REQUEST_METRONOME_JOBS_SUCCESS,
+          data
+        }),
+      error: xhr =>
+        AppDispatcher.handleServerAction({
+          type: REQUEST_METRONOME_JOBS_ERROR,
+          data: xhr.message,
+          xhr
+        })
+    });
+  },
 
   fetchJobDetail(jobID) {
-    RequestUtil.json({
-      url: `${Config.metronomeAPI}/v1/jobs/${jobID}`,
-      data: [
-        { name: "embed", value: "activeRuns" },
-        { name: "embed", value: "history" },
-        { name: "embed", value: "schedules" }
-      ],
-      success(response) {
+    MetronomeClient.fetchJobDetail(jobID).subscribe({
+      next: response =>
```

Comment:
This bit is missing this data part, I guess. I am saying that because I still see a bug on testing the jobs between nuc and this PR, and it is not showing all the previous runs.
```
     data: [	
         { name: "embed", value: "activeRuns" },	
          { name: "embed", value: "history" },
         { name: "embed", value: "schedules" }	
     ],
```

Reply:
You are right, fixing this right now 👍